spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/actuate/AbstractGatewayControllerEndpoint.java

```diff
@@ -145,6 +166,28 @@ public Mono<ResponseEntity<Object>> save(@PathVariable String id, @RequestBody R
 				.switchIfEmpty(Mono.defer(() -> Mono.just(ResponseEntity.badRequest().build())));
 	}
 
+	@PostMapping("/batch/routes")
```

Comment:
Could we keep the endpoint the same `/routes`?

Reply:
I am wondering if Spring will select the right method based on the body of the request...maybe it won't in which case we will need a separate path.

If we need a separate path then I would actually prefer `/routes/batch`.

Reply:
> I am wondering if Spring will select the right method based on the body of the request...maybe it won't in which case we will need a separate path.

We have `@PostMapping("/routes/{id}")` that accepts `RouteDefinition`. I theory we could have another method `@PostMapping("/routes/")` that accepts an array I think. No need for special things.

> If we need a separate path then I would actually prefer `/routes/batch`

That would collide with the current `/routes/{id}`. But if we can have `@PostMapping("/routes/{id}")` and `@PostMapping("/routes/")` , we should be fine.

Reply:
Makes sense